### PR TITLE
Lolcat correctly interprets escape sequences - Fixes #14

### LINKED ISSUE #14: `sl | lolcat` renders incorrectly
[This recording probably sums it up best](https://asciinema.org/a/qe1qIOo2bPHBxQu8nkVMT8Gry).

### PR DESCRIPTION
**Pull request by magnus-ISU**
This allows fun things like `sl | lolcat` and `htop | lolcat`.

They still look sort of weird, but I think there isn't a great way to get that to work, as I think the weirdness depends on the order of escape sequences given to lolhelp more than anything lolhelp is doing.

As for the change in behavior, previously if lolcat encountered an escape sequence, it (attempted to) ignore it without printing it. It was also a bad implementation as it didn't really know what an escape sequence was.
Now, it will echo all escape sequences without changing them (including ones that affect color - in that case it doesn't matter though, because before it prints the next character it change the color again for the lolz).
Stdin is no longer buffered by newlines either.

Restructures quite a bit of the code to use iterators rather than str. I tried to make it clean, but the parsing of escape sequences is also a giant mess which probably can't be written cleaner anyway, unless maybe we found a crate to do it.

Imports `utf8-chars` crate to be able to handle stdin that isn't broken up by newlines.

Does not attempt to gracefully recover from an incorrect escape sequence. However, I've piped a few ncurses programs into it and haven't found anything that causes an error here.

For some reason it doesn't really work with `pacman` progress bars so more work can probably be done.

### CODE REVIEW

**Comment by magnus-ISU** [comment]
Forgot to mention; this PR also makes it so `lolcat -h` can accept other flags, such as `lolcat -hBD`

**Comment by magnus-ISU** [comment]
Added the `-F` flag which forces printing color, even when not printing to a terminal. Bumped the program version to 1.4.0 since the handling of escape sequences is a pretty substantial change imo.

**Comment by magnus-ISU** [comment]
Also, I can take you up on maintaining this going forward. I understand the codebase now, so while I still don't know what more might need to be changed I can definitely get it done if anything comes up.

File: .gitignore [hunk 1]
```diff
@@ -1 +1,2 @@
 target
+Cargo.lock
```
Comment (ur0) [inline .gitignore]: I'm not sure if we should remove the lockfile, as this is a binary crate.

**Comment by ur0** [comment]
@magnus-ISU just a couple of minor notes on the PR, but otherwise this LGTM. Also added you to the collaborators!

**Comment by magnus-ISU** [comment]
Okay, awesome. I will revert deletion of Cargo.lock as you [are correct and I forgot about this](https://doc.rust-lang.org/cargo/faq.html#why-do-binaries-have-cargolock-in-version-control-but-not-libraries). Nice catch.

**Pull request merged**